### PR TITLE
refactor: Updates to Document AI Python Samples

### PR DESCRIPTION
**Pull request by holtskinner**
- Updated Python Samples to `v1` API
- Simplified Quickstart/Online Processing Samples to print out full document text instead of pages/paragraphs
- Updated Batch Processing Sample to Utilize `BatchProcessMetadata` when retrieving output from Google Cloud Storage, rather than the Operation Number
- Added explicit typing where missing
- Moved API calls to separate functions for parsing-focused examples.
- Updated Related Tests
- Added Sample Output JSON Files (Short and Full) with the PDF files that generated them

### Documentation Pages the samples will be embedded in

- Quickstart: https://cloud.google.com/document-ai/docs/process-documents-client-libraries#client-libraries-usage-python
- Online/Batch Processing: https://cloud.google.com/document-ai/docs/send-request
- Handle Response: https://cloud.google.com/document-ai/docs/handle-response

### CODE REVIEW

File: samples/snippets/process_document_quality_sample.py [hunk 1]
```diff
@@ -15,64 +15,83 @@
...
-# project_id= 'YOUR_PROJECT_ID'
-# location = 'YOUR_PROJECT_LOCATION' # Format is 'us' or 'eu'
+# project_id = 'YOUR_PROJECT_ID'
+# location = 'YOUR_PROCESSOR_LOCATION' # Format is 'us' or 'eu'
 # processor_id = 'YOUR_PROCESSOR_ID' # Create processor in Cloud Console
 # file_path = '/path/to/local/pdf'
+# mime_type = 'application/pdf' # Refer to https://cloud.google.com/document-ai/docs/processors-list for supported file types
 
 
 def process_document_quality_sample(
```
Comment (galz10) [inline samples/snippets/process_document_quality_sample.py:31]: Same question about splitting this function into two

File: samples/snippets/process_document_sample.py [hunk 2]
```diff
@@ -15,78 +15,49 @@
...
-# location = 'YOUR_PROJECT_LOCATION' # Format is 'us' or 'eu'
-# processor_id = 'YOUR_PROCESSOR_ID' # Create processor in Cloud Console
+# project_id = 'YOUR_PROJECT_ID'
+# location = 'YOUR_PROCESSOR_LOCATION' # Format is 'us' or 'eu'
+# processor_id = 'YOUR_PROCESSOR_ID' #  Create processor in Cloud Console
 # file_path = '/path/to/local/pdf'
+# mime_type = 'application/pdf' # Refer to https://cloud.google.com/document-ai/docs/processors-list for supported file types
 
 
 def process_document_sample(
```
Comment (galz10) [inline samples/snippets/process_document_sample.py:29]: Same question about removing the printing of text vs printing document.txt

File: samples/snippets/process_document_specialized_sample.py [hunk 3]
```diff
@@ -15,65 +15,85 @@
...
-# project_id= 'YOUR_PROJECT_ID'
-# location = 'YOUR_PROJECT_LOCATION' # Format is 'us' or 'eu'
+# project_id = 'YOUR_PROJECT_ID'
+# location = 'YOUR_PROCESSOR_LOCATION' # Format is 'us' or 'eu'
 # processor_id = 'YOUR_PROCESSOR_ID' # Create processor in Cloud Console
 # file_path = '/path/to/local/pdf'
+# mime_type = 'application/pdf' # Refer to https://cloud.google.com/document-ai/docs/processors-list for supported file types
 
 
 def process_document_specialized_sample(
```
Comment (galz10) [inline samples/snippets/process_document_specialized_sample.py:30]: Same question about splitting this function into two

File: samples/snippets/process_document_splitter_sample.py [hunk 4]
```diff
@@ -15,74 +15,88 @@
...
-# project_id= 'YOUR_PROJECT_ID'
-# location = 'YOUR_PROJECT_LOCATION' # Format is 'us' or 'eu'
+# project_id = 'YOUR_PROJECT_ID'
+# location = 'YOUR_PROCESSOR_LOCATION' # Format is 'us' or 'eu'
 # processor_id = 'YOUR_PROCESSOR_ID' # Create processor in Cloud Console
 # file_path = '/path/to/local/pdf'
+# mime_type = 'application/pdf' # Refer to https://cloud.google.com/document-ai/docs/processors-list for supported file types
 
 
 def process_document_splitter_sample(
```
Comment (galz10) [inline samples/snippets/process_document_splitter_sample.py:31]: Same question about splitting this function into two

File: samples/snippets/quickstart_sample.py [hunk 5]
```diff
@@ -17,73 +17,49 @@
...
-        start_index = (
-            int(segment.start_index)
-            if segment in doc_element.text_anchor.text_segments
-            else 0
-        )
-        end_index = int(segment.end_index)
-        response += document.text[start_index:end_index]
-    return response
+    print("The document contains the following text:")
+    print(document.text)
```
Comment (galz10) [inline samples/snippets/quickstart_sample.py:62]: Same question about document.txt vs get_text

File: samples/snippets/batch_process_documents_sample.py [hunk 6]
```diff
@@ -16,118 +16,136 @@
...
+    gcs_documents = documentai.GcsDocuments(documents=[gcs_document])
     input_config = documentai.BatchDocumentsInputConfig(gcs_documents=gcs_documents)
 
-    # Where to write results
-    output_config = documentai.DocumentOutputConfig(
-        gcs_output_config={"gcs_uri": destination_uri}
+    # NOTE: Alternatively, specify a GCS URI Prefix to process an entire directory
+    #
+    # gcs_input_uri = "gs://bucket/directory/"
+    # gcs_prefix = documentai.GcsPrefix(gcs_uri_prefix=GCS_INPUT_PREFIX)
```
[the remark this answers is not in the document]
Comment (holtskinner) [reply samples/snippets/batch_process_documents_sample.py:60]: Good catch, resolved

File: samples/snippets/process_document_ocr_sample.py [hunk 7]
```diff
@@ -15,99 +15,113 @@
...
-# project_id= 'YOUR_PROJECT_ID'
-# location = 'YOUR_PROJECT_LOCATION' # Format is 'us' or 'eu'
+# project_id = 'YOUR_PROJECT_ID'
+# location = 'YOUR_PROCESSOR_LOCATION' # Format is 'us' or 'eu'
 # processor_id = 'YOUR_PROCESSOR_ID' # Create processor in Cloud Console
 # file_path = '/path/to/local/pdf'
+# mime_type = 'application/pdf' # Refer to https://cloud.google.com/document-ai/docs/processors-list for supported file types
 
 
 def process_document_ocr_sample(
```
[the remark this answers is not in the document]
Comment (holtskinner) [reply samples/snippets/process_document_ocr_sample.py:31]: Similar response to above, this one is focused on extracting out the OCR data from the Document Object, not making the processing request.

File: samples/snippets/process_document_quality_sample.py (hunk 1 above)
Comment (holtskinner) [reply samples/snippets/process_document_quality_sample.py:31]: Similar answer, this example is focused on handling the processing response, not making the request

File: samples/snippets/process_document_sample.py (hunk 2 above)
Comment (holtskinner) [reply samples/snippets/process_document_sample.py:29]: I removed the page/paragraph extraction from this example to focus this Online Processing example on making the processing request, not parsing the response.

printing document.text shows that the document was processed successfully, but the sample doesn't need to go into depth about the response structure. The OCR example is designed to show all of the fields that can be extracted from the document response

File: samples/snippets/process_document_specialized_sample.py (hunk 3 above)
Comment (holtskinner) [reply samples/snippets/process_document_specialized_sample.py:30]: Similar answer, this example is focused on parsing the response for a specialized document type

File: samples/snippets/process_document_splitter_sample.py (hunk 4 above)
Comment (holtskinner) [reply samples/snippets/process_document_splitter_sample.py:31]: Similar answer, this example is focused on extracting the document object information from a splitter/classifier processor

File: samples/snippets/quickstart_sample.py (hunk 5 above)
Comment (holtskinner) [reply samples/snippets/quickstart_sample.py:62]: This example is exactly the same as the online processing example. Focused on making the processing request, not parsing the response. The OCR/Form parsing/etc examples are focused on handling the response, not constructing the request

**Review by holtskinner — COMMENTED** [review]
Addressed Comments

General discussion of structure

The Process_document and batch_process_document samples are focused on constructing/making the Document AI API request, the rest of the samples are focused on handling the Document Response for different processor types, so I brought out the process document request into a separate function to bring the focus on the parsing. This is also why I removed the pages/paragraphs handling from the quickstart/online processing samples.

These are the Docs Pages the samples will be shown in, for context

- Quickstart: https://cloud.google.com/document-ai/docs/process-documents-client-libraries#client-libraries-usage-python
- Online/Batch Processing: https://cloud.google.com/document-ai/docs/send-request
- Handle Response: https://cloud.google.com/document-ai/docs/handle-response

File: samples/snippets/batch_process_documents_sample.py [hunk 8]
```diff
@@ -16,118 +16,136 @@
...
+            document = documentai.Document.from_json(
+                blob.download_as_bytes(), ignore_unknown_fields=True
+            )
 
             # For a full list of Document object attributes, please reference this page:
-            # https://cloud.google.com/document-ai/docs/reference/rpc/google.cloud.documentai.v1beta3#document
+            # https://cloud.google.com/python/docs/reference/documentai/latest/google.cloud.documentai_v1.types.Document
 
             # Read the text recognition output from the processor
-            for page in document.pages:
```
[the remark this answers is not in the document]
Comment (holtskinner) [reply samples/snippets/batch_process_documents_sample.py:99]: I removed the page/paragraph extraction from this example to focus this Online Processing example on making the processing request, not parsing the response.

printing document.text shows that the document was processed successfully, but the sample doesn't need to go into depth about the response structure. The OCR example is designed to show all of the fields that can be extracted from the document response

**Review by matthewayne — APPROVED** [review]

**Review by parthea — CHANGES_REQUESTED** [review]
Please could you take a look at the failing samples tests?

**Comment by holtskinner** [comment]
Ok, it looks like the Kokoro Unit Test only failed because of a quota error on the unit test project. Would it be possible to get a quota increase request on the test project if we're going to run multiple batch requests simultaneously?
Screen Shot 2022-07-26 at 12 02 33 PM

File: samples/snippets/requirements.txt [hunk 9]
```diff
@@ -1,2 +1,3 @@
-google-cloud-documentai==1.4.2
-google-cloud-storage==2.3.0
+google-api-core==2.8.2
```
Comment (parthea) [inline samples/snippets/requirements.txt:1]:
```suggestion
```
google-api-core is already a dependency of this library.

**Review by parthea — APPROVED** [review]
Added minor observations, otherwise LGTM

**Pull request merged**